configs/makeIngestSpec.py

```diff
@@ -0,0 +1,136 @@
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import yaml
+import json
+import os
+
+def transform_ingest_json(template_filename, output_filename, schema_columns):
+    """
+    Reads in a json file from template_filename, and inserts into the schema
+    section a list of the columns provided in schema_columns. This replaces the first
+    zero-length entry in the template file's schema entry, e.g. an empty string, while
+    preserving any additional columns present in the template.
+    """
+
+    try:
+        with open(template_filename) as f:
+            input_json = json.load(f)
+    except FileNotFoundError:
+        print("Could not locate template file {:s}, skipping.".format(template_filename))
+        return
+
+    output_json = input_json.copy()
+
+    for n in range(len(output_json["schema"])):
+        if(len(output_json["schema"][n]) == 0):
+            output_json["schema"][n:n+1] = schema_columns
+        break
+
+    print("Writing out {:s}".format(output_filename))
+    with open(output_filename, "w") as f:
+        json.dump(output_json, f, indent=4)
+
+def transform_partitioner_json(template_filename, output_filename, schema_columns):
+    """
+    Reads in a json file from template_filename, and inserts into the schema
+    section a list of the columns provided in schema_columns. This replaces the first
+    zero-length entry in the template file's schema entry, e.g. an empty string, while
+    preserving any additional columns present in the template.
+    """
+
+    try:
+        with open(template_filename) as f:
+            input_json = json.load(f)
+    except FileNotFoundError:
+        print("Could not locate template file {:s}, skipping.".format(template_filename))
+        return
+
+    output_json = input_json.copy()
+
+    output_field = output_json["out"]["csv"]["field"]
+    for n in range(len(output_field)):
+        if(len(output_field[n]) == 0):
+            output_field[n:n+1] = schema_columns
+
+    input_field = output_json["in"]["csv"]["field"]
+    for n in range(len(input_field)):
+        if(len(input_field[n]) == 0):
+            input_field[n:n+1] = schema_columns
```

Comment:
I think it may be a bit better if looked for an entry like {"name":"","type":"SCHEMA_PLACEHOLDER"} instead of a zero length field. I don't think it makes that much difference, but being explicit may be a bit clearer what is happening in the json file.  Same applies to transform_ingest_json.